src/main/java/core/basesyntax/service/operations/PurchaseOperation.java

```diff
@@ -0,0 +1,11 @@
+package core.basesyntax.service.operations;
+
+import core.basesyntax.model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        storageFruits.put(transaction.getFruit(), storageFruits
+                .getOrDefault(transaction.getFruit(), 0) - transaction.getQuantity());
+    }
+}
```

Comment:
The class `PurchaseOperation` should not directly access `storageFruits`. It violates the encapsulation principle. Instead, it should interact with the storage through a service or DAO layer that manages the access.